src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -71,6 +71,16 @@ def __init__(
             new_config["steps_offset"] = 1
             scheduler._internal_dict = FrozenDict(new_config)
 
+        if safety_checker is None:
+            logger.warn(
+                f"You have disabed the safety checker for {self.__class__} by passing `safety_checker=None`.Please"
+                " make sure you have very good reasons for this and have considered the consequences of doing so.The"
+                " `diffusers` team does not recommend disabling the safety under ANY circumstances and strongly"
+                " suggests to not disable the `safety_checker` by NOT passing `safety_checker=None` to"
+                " `from_pretrained`. For more information, please have a look at"
+                " https://github.com/huggingface/diffusers/pull/254"
+            )
+
```

Comment:
I think referencing another PR dilutes the message. I'd propose something like:

> The `diffusers` team and Hugging Face strongly recommend to keep the safety filter enabled in all circumstances. Please, make sure you abide to the conditions of the Stable Diffusion license and do not expose unfiltered results in services or applications open to the public.

I think it's worthwhile to make this as clear as we can, maybe @natolambert, @mmitchellai, @yjernite can provide better wording.

Reply:
I would even say: 

> Ensure that you abide to the conditions of the Stable Diffusion license and do not expose unfiltered results in services or applications open to the public. Both the diffusers team and Hugging Face strongly recommend to keep the safety filter enabled in all public facing circumstances, disabling it only for use-cases that involve analyzing network behavior or auditing its results.


I mean, I think proposing a use case when removing the safety filter is acceptable kind of justifies why we are proposing this PR to remove it.